### PR TITLE
Add CRNN Text Recognition Demo

### PR DESCRIPTION
**Pull request by avbelova**
@IRDonch @Wovchena please review.

### CODE REVIEW

File: demos/python_demos/crnn_text_recognition_demo/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# CRNN Text Recognition Python* Demo
```
Comment (Wovchena) [inline demos/python_demos/crnn_text_recognition_demo/README.md]: Please, add the demo to the `demos/README.md`'s lsit.

File: demos/python_demos/crnn_text_recognition_demo/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## See Also
+
+* [Using Open Model Zoo demos](../../README.md)
+* [Model Optimizer](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html)
+* [Model Downloader](../../../tools/downloader/README.md)
+
+
+
+
```
Comment (Wovchena) [inline demos/python_demos/crnn_text_recognition_demo/README.md]: Extra lines

File: demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                      required=True, type=str, metavar='"<path>"')
+    args.add_argument('-d', '--device',
+                      help='Optional. Specify the target device to infer on: CPU, GPU, FPGA, HDDL or MYRIAD. '
+                           'The demo will look for a suitable plugin for device specified '
+                           '(by default, it is CPU).',
+                           default='CPU', type=str, metavar='"<device>"')
+    args.add_argument('-l', '--cpu_extension',
+                      help='Required for CPU custom layers. '
+                           'Absolute path to a shared library with the kernels implementation.',
+                      default=None, type=str, metavar='"<absolute_path>"')
```
Comment (Wovchena) [inline demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py]: Why is it `"<absolute_path>" ` for `-l` in `metavar`, while for others it's just `"<path>"`?

File: demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    if args.cpu_extension and 'CPU' in args.device:
+        ie.add_extension(args.cpu_extension, 'CPU')
+        
+    if 'CPU' in args.device:
+        supported_layers = ie.query_network(net, 'CPU')
+        not_supported_layers = [l for l in net.layers.keys() if l not in supported_layers]
+        if len(not_supported_layers) != 0:
+            log.error('Following layers are not supported by the plugin for specified device {}:\n {}'.
+                      format(args.device, ', '.join(not_supported_layers)))
+            log.error("Please try to specify cpu extensions library path in sample's command line parameters using -l "
```
Comment (Wovchena) [inline demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py]: It's demo, but not sample :)

File: demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    input_blob = next(iter(net.inputs))
+    n, c, h, w = net.inputs[input_blob].shape
+    image = cv2.resize(input_image, (w, h))
+    image = image.transpose((2, 0, 1))
+    image = np.expand_dims(image,axis=0)
+
+    #run inference
+    log.info('Starting inference...')
+    res = exec_net.infer(inputs={input_blob: image})
+    out_blob = next(iter(net.outputs))
```
Comment (Wovchena) [inline demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py]: You should check that the model input is of expected size: number of inputs, dimensions of a blob. The same is for `net.outputs`.

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py (hunk 3 above)
Comment (Wovchena) [reply demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py]: `metavar`'s value should also be updated.

File: demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    msg='Detected text is "{}"'.format("".join(word))
+    log.info(msg)
+    cv2.imshow(msg,input_image)
+    cv2.waitKey()
+    log.info('Demo finished successfully')
+    log.info("This demo is an API example, for any performance measurements please use the dedicated benchmark_app tool "
+             "from the openVINO toolkit\n")
+
+if __name__ == '__main__':
+    sys.exit(main() or 0)
```
Comment (Wovchena) [inline demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py]: I don't think you want to drop the newline at end of file.

**Comment by Wovchena** [comment]
Adding the model to downloader would require adding tests for the demo which would require adding `--no_show` key disabling `imshow` and `waitKey`.

File: demos/README.md [hunk 7]
```diff
@@ -6,6 +6,7 @@ The Open Model Zoo demo applications are console applications that demonstrate h
 The Open Model Zoo includes the following demos:
 
 - [Action Recognition Python* Demo](./python_demos/action_recognition/README.md) - Demo application for Action Recognition algorithm, which classifies actions that are being performed on input video.
+- [CRNN Text Recognition Python* Demo] (./python_demos/crnn_text_recognition_demo/README.md) - Demo application for English text recognition based on CRNN (Convolutional Reccurent Neural Network) model.
```
Comment (ngaloppo) [inline demos/README.md]:
```suggestion
- [CRNN Text Recognition Python* Demo] (./python_demos/crnn_text_recognition_demo/README.md) - Demo application for English text recognition based on CRNN (Convolutional Recurrent Neural Network) model.
```

Comment (ngaloppo) [reply demos/README.md]: Fixed typo.

**Comment by IRDonch** [comment]
I noticed that the interface of the CRNN model is nearly the same as that of our `text-recognition-0012` model. The only differences are:

1. The input is color rather than gray.
2. The maximum number of characters is different (30 vs 25).

Given this, is it really a good idea to create a demo just for this model? I think it would be better to adapt our existing `text_detection_demo` to support this model instead.

**Comment by IRDonch** [comment]
Closing due to inactivity.

**Pull request closed without merging**